### PR TITLE
Move removing unneeded autoprovisioned node groups to node group manager

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -338,12 +338,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 
 			// We want to delete unneeded Node Groups only if there was no recent scale up,
 			// and there is no current delete in progress and there was no recent errors.
-			if a.AutoscalingContext.NodeAutoprovisioningEnabled {
-				err := cleanUpNodeAutoprovisionedGroups(a.AutoscalingContext.CloudProvider, a.AutoscalingContext.LogRecorder)
-				if err != nil {
-					glog.Warningf("Failed to clean up unneeded node groups: %v", err)
-				}
-			}
+			a.processors.NodeGroupManager.RemoveUnneededNodeGroups(autoscalingContext)
```
Comment (losipiuk) [inline cluster-autoscaler/core/static_autoscaler.go]: Is NodeGroupManager always set here?

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/static_autoscaler.go]: yes

**Pull request merged**